app/assets/javascripts/discourse/app/models/composer.js

```diff
@@ -442,10 +442,14 @@ const Composer = RestModel.extend({
 
   @discourseComputed("canCategorize", "categoryId")
   requiredCategoryMissing(canCategorize, categoryId) {
+    const hasTopicTemplates = this.site.categories.some(
```

Comment:
I don't think we should run this check every single time the property is computed (looks like whenever the current category is changed). Can you memoize it in the model?